MapboxNavigation/FeedbackViewController.swift

```diff
@@ -70,6 +115,64 @@ class FeedbackViewController: UIViewController {
         dismissFeedback()
     }
     
+    func didTapCell(_ sender: UIGestureRecognizer) {
+        guard sender.state == .began || sender.state == .ended else { return }
+        
+        abortAutodismiss()
+        
+        let touchLocation = sender.location(in: self.collectionView)
+        
+        if let indexPath = self.collectionView.indexPathForItem(at: touchLocation) {
+            // get the cell at indexPath (the one you long pressed)
+            let cell = self.collectionView.cellForItem(at: indexPath) as! FeedbackCollectionViewCell
+            cell.layer.add(pulsingAnimation, forKey: "animateScale")
+            
+            activeFeedbackItem = sections[indexPath.section][indexPath.row]
+            
+            
+            if audioRecorder == nil {
+                startRecording()
+            }
+            
+            if sender.state == .ended {
+                finishRecording()
+            }
+        }
+    }
+    
+    func audioRecorderDidFinishRecording(_ recorder: AVAudioRecorder, successfully flag: Bool) {
+        if let fileData = NSData(contentsOfFile: recorder.url.path) as Data? {
+            activeFeedbackItem!.audio = fileData
+            sendFeedbackHandler?(activeFeedbackItem!)
+        }
+    }
+    
+    
+    func startRecording() {
+        guard let audioFile = FileManager().cacheAudioFileLocation else { return }
+        
+        let settings = [
+            AVFormatIDKey: Int(kAudioFormatMPEG4AAC),
+            AVSampleRateKey: 12000,
+            AVNumberOfChannelsKey: 1,
+            AVEncoderAudioQualityKey: AVAudioQuality.low.rawValue
+        ]
+        
+        do {
+            audioRecorder = try AVAudioRecorder(url: audioFile, settings: settings)
+            audioRecorder?.delegate = self
```

Comment:
I think it might not be necessary here to set the delegate.

Reply:
Correct, we can remove this and the conformance to AVAudioRecorderDelegate.